bip-opt-in-rbf.mediawiki

```diff
@@ -0,0 +1,187 @@
+<pre>
+  BIP: bip-harding-opt-in-replace-by-fee
+  Title: Opt-in Full Replace-by-Fee Signaling
+  Author: David A. Harding <dave@dtrt.org>, Peter Todd <pete@petertodd.org>
+  Status: Draft
+  Type: Standards Track
+  Created: 2015-12-04
+</pre>
+
+==Abstract==
+
+Many nodes today will not replace any transaction in their mempool with
+another transaction that spends the same inputs, making it difficult for
+spenders to adjust their previously-sent transactions to deal with
+unexpected confirmation delays or to perform other useful replacements.
+
+The opt-in full Replace-by-Fee (opt-in full-RBF) signaling policy
+described here allows spenders to add a signal to a transaction indicating
+that they want to be able to replace that transaction in the future.
+In response to this signal,
+
+* Nodes may allow transactions containing this signal to be replaced in their mempools.
+
+* The recipient or recipients of a transaction containing this signal may choose not to treat it as payment until it has been confirmed, eliminating the risk that the spender will use allowed replacements to defraud them.
```

Comment:
There aren't any incentive compatible partial block proposals yet; not relevant to the current ecosystem.

On 19 December 2015 09:54:25 GMT-08:00, Aaron Voisine <notifications@github.com> wrote:
>> +
>> +==Abstract==
>> +
>> +Many nodes today will not replace any transaction in their mempool
>with
>> +another transaction that spends the same inputs, making it difficult
>for
>> +spenders to adjust their previously-sent transactions to deal with
>> +unexpected confirmation delays or to perform other useful
>replacements.
>> +
>> +The opt-in full Replace-by-Fee (opt-in full-RBF) signaling policy
>> +described here allows spenders to add a signal to a transaction
>indicating
>> +that they want to be able to replace that transaction in the future.
>> +In response to this signal,
>> +
>> +* Nodes may allow transactions containing this signal to be replaced
>in their mempools.
>> +
>> +* The recipient or recipients of a transaction containing this
>signal may choose not to treat it as payment until it has been
>confirmed, eliminating the risk that the spender will use allowed
>replacements to defraud them.
>
>What about by broadcasting partial block solutions? This is a
>decentralized way to indicate how much hashing power is being expended
>attempting to mine any given transaction.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/bitcoin/bips/pull/261/files#r48095057